doc/lsst.ap.pipe/pipeline-tutorial.rst

```diff
@@ -0,0 +1,255 @@
+in ``ap_pipe/bin/ap_pipe.py``.
+
+To process your ingested data, run
+
+.. prompt:: bash
+   
+   ap_pipe.py repo --calib repo/calibs --rerun processed --id visit=123456 ccdnum=42 filter=g
+
+In this case, a ``processed`` directory will be created within
+``repo/rerun`` and the results will be written there.
+Using coadds as templates makes `reruns 
+<https://pipelines.lsst.io/getting-started/processccd.html#aside-reruns-and-output-butler-repositories>`_
+useful for many workflows,
+because it is possible to run ``ap_pipe`` many times while using the same
+repository of coadd templates as both an input and output location on disk.
+
+This example command only processes observations that have a ``dataId``
+corresponding to visit 123456 and ccdnum 42 in with a filter called g.
+You must specify a ``dataId`` string or no data will be processed.
+`lsst.ap.pipe` supports ``dataId`` parsing, e.g., ``ccdnum=3^6..12`` will process
+``ccdnums`` 3, 6, 7, 8, 9, 10, 11, and 12.
+
+.. note::
+
+   Until a resolution for `DM-12672 <https://jira.lsstcorp.org/browse/DM-12672>`_
+   is found, you should include a filter in the ``dataId`` string for
+   ``ap_pipe`` to run successfully.
+
+If you prefer to have a standalone output repository and specify the template location
+explicitly, you may instead run
+
+.. prompt:: bash
+
+   ap_pipe.py repo --calib repo/calibs --output path/to/put/processed/data/in --id visit=123456 ccdnum=42 filter=g --template path/to/templates
+
+In this case, the output directory will be created if it does not already exist.
+If you omit the ``--template`` flag, ``ap_pipe`` will assume the templates are
+somewhere in ``repo``.
+
+
+.. _section-expected-outputs:
+
+Expected outputs
+================
+
+If you used the rerun option above, the output from ``ap_pipe`` should be
+written out as follows
+
+.. code-block:: none
+
+   repo/
+      rerun/
+         processed/
+            repositoryCfg.yaml
+            association.db   <--- the Prompt Products Database with DIAObjects
+            config/
+            deepDiff/
+               v123456/   <--- difference images and DIASource tables are in here
+            schema/
+            123456/   <--- all other processed data products are in here (calexps etc.)
+
+Of course, to inspect this data with the Butler, you don't need to know
+where it lives on disk. You should instead instantiate a Butler within python
+in the ``processed`` directory and access the data products that way.
+
+For example, in python
+
+.. code-block:: python
+   
+   import lsst.daf.persistence as dafPersist
+   butler = dafPersist.Butler('repo')
+   dataId = {'visit': 123456, 'ccdnum': 42, 'filter': 'g'}
+   calexp = butler.get('calexp', dataId=dataId)
+   diffim = butler.get('deepDiff_differenceExp', dataId=dataId)
+   diaSourceTable = butler.get('deepDiff_diaSrc', dataId=dataId)
+
+
+.. _section-configuration-options:
+
+Configuration options
+=====================
```

Comment:
This is kind of a confusing section, because it combines different sorts of things. Why not move the subsections to top-level sections of their own? (I guess "previewing data IDs" might be easiest to find if it went in the main tutorial?)